### ISSUE TITLE
[Refactor][framework] Drop temporal support

### ISSUE THREAD

**Issue by klesh** [apache/devlake#5939]
## What and why to refactor
We once believed it would be beneficial to support collecting data distributedly via Temporal and #1513 had been done for more than a year. However, it is never been used by anyone due to the following limitation:

- Most data source has rate limit which makes distributed collection practically useless
- It is quite expensive to deploy such a distributed system because workers must be deployed into different machines

To sum up, `temporal` support is not practical for us except for adding complexity to the codebase which makes refactoring harder.

## Describe the solution you'd like
Drop `Temporal` support, remove related code and simplify the pipeline execution logic

**Comment by keon94** [comment]
I can take this one @klesh
